drivers/accel/adxl355/adxl355.h

```diff
@@ -0,0 +1,303 @@
+#define ADXL355_OFFSET_X     (ADXL355_ADDR(0x1E) | SET_ADXL355_TRANSF_LEN(2))
+#define ADXL355_OFFSET_Y     (ADXL355_ADDR(0x20) | SET_ADXL355_TRANSF_LEN(2))
+#define ADXL355_OFFSET_Z     (ADXL355_ADDR(0x22) | SET_ADXL355_TRANSF_LEN(2))
+#define ADXL355_ACT_EN       (ADXL355_ADDR(0x24) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x00))
+#define ADXL355_ACT_THRESH   (ADXL355_ADDR(0x25) | SET_ADXL355_TRANSF_LEN(2))
+#define ADXL355_ACT_COUNT    (ADXL355_ADDR(0x27) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x01))
+#define ADXL355_FILTER       (ADXL355_ADDR(0x28) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x00))
+#define ADXL355_FIFO_SAMPLES (ADXL355_ADDR(0x29) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x60))
+#define ADXL355_INT_MAP      (ADXL355_ADDR(0x2A) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x00))
+#define ADXL355_SYNC         (ADXL355_ADDR(0x2B) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x00))
+#define ADXL355_RANGE        (ADXL355_ADDR(0x2C) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x81))
+#define ADXL355_POWER_CTL    (ADXL355_ADDR(0x2D) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x01))
+#define ADXL355_SELF_TEST    (ADXL355_ADDR(0x2E) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x00))
+#define ADXL355_RESET        (ADXL355_ADDR(0x2F) | SET_ADXL355_TRANSF_LEN(1) | SET_ADXL355_RESET_VAL(0x00))
+
+#define ADXL355_SHADOW_REGISTER_BASE_ADDR (ADXL355_ADDR(0x50) | SET_ADXL355_TRANSF_LEN(5))
+#define ADXL355_MAX_FIFO_SAMPLES_VAL 0x60
+
+#define ADXL355_SELF_TEST_TRIGGER_VAL 0x03
+
+#define ADXL355_ACC_SCALE_FACTOR 0.00003824593
+
+#define ADXL355_TEMP_SCALE_FACTOR -2111.25
+#define ADXL355_TEMP_OFFSET -0.11049723
+
+#define ADXL355_RESET_CODE 0x52
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+enum adxl355_op_mode {
+	ADXL355_MEASUREMENT_TEMP_ON_DRDY_ON = 0,
+	ADXL355_STANDBY_TEMP_ON_DRDY_ON = 1,
+	ADXL355_MEASUREMENT_TEMP_OFF_DRDY_ON = 2,
+	ADXL355_STANDBY_TEMP_OFF_DRDY_ON = 3,
+	ADXL355_MEASUREMENT_TEMP_ON_DRDY_OFF = 4,
+	ADXL355_STANDBY_TEMP_ON_DRDY_OFF = 5,
+	ADXL355_MEASUREMENT_TEMP_OFF_DRDY_OFF = 6,
+	ADXL355_STANDBY_TEMP_OFF_DRDY_OFF = 7
+
+};
+
+enum adxl355_comm_type {
+	ADXL355_SPI_COMM,
+	ADXL355_I2C_COMM
+};
+
+enum adxl355_hpf_corner {
+	ADXL355_HPF_OFF,
+	ADXL355_HPF_24_7,
+	ADXL355_HPF_6_2084,
+	ADXL355_HPF_1_5545,
+	ADXL355_HPF_0_3862,
+	ADXL355_HPF_0_0954,
+	ADXL355_HPF_0_0238
+};
+
+enum adxl355_odr_lpf {
+	ADXL355_ODR_4000HZ,
+	ADXL355_ODR_2000HZ,
+	ADXL355_ODR_1000HZ,
+	ADXL355_ODR_500HZ,
+	ADXL355_ODR_250HZ,
+	ADXL355_ODR_125HZ,
+	ADXL355_ODR_62_5HZ,
+	ADXL355_ODR_31_25HZ,
+	ADXL355_ODR_15_625HZ,
+	ADXL355_ODR_7_813HZ,
+	ADXL355_ODR_3_906HZ
+};
+
+enum adxl355_range {
+	ADXL355_RANGE_INVALID,
```

Comment:
This is a good habit, having invalid values in enums ;)
But we tend to prefer the other method where we match these enums to register field values.
So if the device register where range is set defaults to value zero (which means range 2G), then we simply start the enum at ADXL355_RANGE_255 which takes this default  zero value.

The advantage is simpler code, with some implicit defaults, therefore, you can also remove:
```

	// Default range is set
	dev->range = ADXL355_RANGE_2G;

	// Default value for FIFO SAMPLES
	dev->fifo_samples =  ADXL355_MAX_FIFO_SAMPLES_VAL;

	// Default value for POWER_CTL
	dev->op_mode = ADXL355_STANDBY_TEMP_ON_DRDY_ON;
```

Reply:
I still have to initialize the driver with the value of range, just in case the user does not set the range manually, because dev->range is actually being used inside adxl355_accel_conversion function. 
The same goes for dev->op_mode which is used in adxl355_set_odr_lpf.
I would also like to keep the initialization of dev->fifo_samples (which is different from 0) for debugging purposes, such that the user can see the actual configuration of the device.  

However, some improvement could be made and this is how I modified the code section you mentioned:

```
        // Default range is set
	dev->range = (enum adxl355_range)(GET_ADXL355_RESET_VAL(
			ADXL355_RANGE) & GENMASK(1,0));
	// Default value for FIFO SAMPLES
	dev->fifo_samples = GET_ADXL355_RESET_VAL(ADXL355_FIFO_SAMPLES);

	// Default value for POWER_CTL
	dev->op_mode = (enum adxl355_op_mode)(GET_ADXL355_RESET_VAL(ADXL355_POWER_CTL));
```

Thank you for your feedback!